pkg/providers/vsphere/vmprovider.go

```diff
@@ -459,6 +461,82 @@ func (vs *vSphereVMProvider) GetItemFromInventoryByName(
 	return vm, nil
 }
 
+func (vs *vSphereVMProvider) GetCloneTasksForVM(
```

Comment:
Looks like this function is generic enough that not only works for clone task? Could it be renamed to something like `GetTasksForVMByDescriptionID`?